Frontends/CMakeLists.txt

```diff
@@ -51,6 +51,8 @@ if(BUILD_CSBEATS)
         set_source_files_properties(${FLEX_csbeats_SCANNER_OUTPUTS} PROPERTIES
         COMPILE_FLAGS -DYY_NO_UNISTD_H)
       make_executable(csbeats "${csbeats_SRCS}" "")
+    elseif(HAIKU)
+      make_executable(csbeats "${csbeats_SRCS}" "")
```

Comment:
We've seen this pattern a few times. I wonder if we shouldn't instead in the main CMakeLists.txt do this:

```cmake
# Some platforms have folded libm into libc so no need to link them
if (HAIKU OR MSVC) 
  set(MATH_LIBRARY "")
else()
  set(MATH_LIBRARY "m")
endif()
```

And then later use that instead of adding a large amount of ifs for each target that requires math functions. We even have a special check in `make_plugin` (ugh)

Reply:
I agree with Felipe; keeping cmake simple is a worthy end.